.github/workflows/ci-core.yml

```diff
@@ -1,7 +1,8 @@
 name: CI Core
+run-name: CI Core ${{ inputs.distinct_run_name && inputs.distinct_run_name || '' }}
```

Comment:
The distinct_run_name usage is required here to properly find the running workflow after it has kicked off. Without this there is a chance we grab the wrong one when the repo is running workflows for several PRs around the same time. Github doesn't provide a built in way to get this in the api when doing a workflow_dispatch so we have to work around that.